AsyncWebServerLogging doesnt compile on ESP8266

### Platform

ESP8266

### IDE / Tooling

PlatformIO

### What happened?

First encountered this while compiling [this](https://github.com/tqlabs-in/dimled-mini/blob/main/esphome/device/base.yaml) on esphome@dev. 

Main cause is that I disable serial logging on esp8266.

### Stack Trace

Useful Snippet
```
mini/ESPAsyncWebServer/src/AsyncWebSocket.h:32,
                 from .piolibdeps/dimled-mini/ESPAsyncWebServer/src/ESPAsyncWebServer.h:1589,
                 from src/esphome/components/web_server_base/web_server_base.h:15,
                 from src/esphome/components/captive_portal/captive_portal.h:13,
                 from src/esphome/components/captive_portal/captive_portal.cpp:1:
.piolibdeps/dimled-mini/ESPAsyncWebServer/src/AsyncWebSocket.h: In constructor 'AsyncWebSocketControl::AsyncWebSocketControl(uint8_t, const uint8_t*, size_t, bool)':
.piolibdeps/dimled-mini/ESPAsyncWebServer/src/AsyncWebServerLogging.h:101:37: error: 'Serial' was not declared in this scope
  101 | #define async_ws_log_e(format, ...) Serial.printf_P(PSTR("E async_ws %d: " format "\n"), __LINE__, ##__VA_ARGS__)
      |                                     ^~~~~~
.piolibdeps/dimled-mini/ESPAsyncWebServer/src/AsyncWebSocket.h:86:9: note: in expansion of macro 'async_ws_log_e'
   86 |         async_ws_log_e("Failed to allocate");
      |         ^~~~~~~~~~~~~~
Compiling .pioenvs/dimled-mini/src/esphome/components/ddp/ddp_light_effect.cpp.o
Compiling .pioenvs/dimled-mini/src/esphome/components/ddp/ddp_light_effect_base.cpp.o
*** [.pioenvs/dimled-mini/src/esphome/components/captive_portal/captive_portal.cpp.o] Error 1
========================= [FAILED] Took 34.56 seconds =========================
```

Full Logs: https://github.com/tqlabs-in/dimled-mini/actions/runs/21462827254/job/61831213713

### Minimal Reproductible Example (MRE)

Actual code is Internally generated via esphome. But this is the cause I guess

```
web_server:

logger:
  baud_rate: 0
```

### I confirm that:

- [x] I have read the documentation.
- [x] I have searched for similar discussions.
- [x] I have searched for similar issues.
- [x] I have looked at the examples.
- [x] I have upgraded to the lasted version of ESPAsyncWebServer (and AsyncTCP for ESP32).

--- a/src/AsyncWebServerLogging.h
+++ b/src/AsyncWebServerLogging.h
@@ -104,9 +104,8 @@
 #define _ASYNC_WS_LOG(level, format, ...)                               \
   do {                                                                  \
     static const char __fmt[] PROGMEM = "%c async_ws %d: " format "\n"; \
-    char __buf[64];                                                     \
-    strncpy_P(__buf, __fmt, sizeof(__buf) - 1);                         \
-    __buf[sizeof(__buf) - 1] = '\0';                                    \
+    char __buf[sizeof(__fmt)];                                          \
+    strcpy_P(__buf, __fmt);                                             \
     ets_printf(__buf, level, __LINE__, ##__VA_ARGS__);                  \
   } while (0)
 // error

